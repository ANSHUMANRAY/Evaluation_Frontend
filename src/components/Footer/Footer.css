.themeColor{
    height: 30px;
    width: 30px;
    border-radius: 10px;
}

.footer{
    display: flex;
    justify-content: space-around;
    align-items: center;
    padding: 10px;
    bottom: 0;
}