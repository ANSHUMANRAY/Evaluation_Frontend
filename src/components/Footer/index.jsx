import React from 'react';
import axios from 'axios';
import { ThemeContext } from '../../contexts/themeContext';
import './Footer.css';

export default function Footer() {
  const {theme, setTheme } = React.useContext(ThemeContext);
  const [themeArray, setThemeArray] = React.useState([]);
  React.useEffect(() => {
    axios.get('http://localhost:8000/api/themes')
      .then((response) => {
        const {themes, preferredTheme} = response.data;
        setThemeArray(themes);
        const theme = (themes.find((theme) => theme.id === preferredTheme));
        setTheme(theme.colorHexCode);
      });
  }, []);

  const handleClick = (id) => {
    axios.put('http://localhost:8000/api/themes', {preferredThemeId: id})
      .then(() =>{
        const tempTheme = themeArray.find((theme) => theme.id === id);
        setTheme(tempTheme.colorHexCode);
      });
  };
  return (
    <div style={{backgroundColor:`${theme}`}} className="footer">
      {themeArray.map((theme) => {
        return (
          <div onClick={() => {handleClick(theme.id);}} key={theme.id} style={{backgroundColor:`${theme.colorHexCode}`}} className="themeColor"></div>
        );
      })}
    </div>);
}