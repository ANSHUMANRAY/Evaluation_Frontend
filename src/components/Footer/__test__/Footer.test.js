import React from 'react';
import { render } from '@testing-library/react';
import Footer from '../index';
import axios from 'axios';
import { ThemeContext } from '../../../contexts/themeContext';
import { mockThemes } from '../../../mocks/mockThemes';

jest.mock('axios');

describe('Footer', () => {
  it('should render correctly', () => {
    axios.get.mockResolvedValue({ data: mockThemes });
    const { container } = render(
      <ThemeContext.Provider value={{ theme: 'black', setTheme: jest.fn() }}>
        <Footer />
      </ThemeContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
