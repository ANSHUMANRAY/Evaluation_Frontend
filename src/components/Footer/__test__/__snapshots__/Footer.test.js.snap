// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<div>
  <div
    class="footer"
    style="background-color: black;"
  />
</div>
`;
